### ISSUE TITLE
Widget instance created twice without need

### ISSUE THREAD

**Issue by micropolimi** [GiorgiaTortora/napari-roi-registration#2]
The instance of the widgets created [here](https://github.com/GiorgiaTortora/napari-roi-registration/blob/a60175388f223e74a25b73927fa8855423452d21/src/napari_roi_registration/_widget.py#L442) might be unnecessary as it is the plugin itself to create the instance.

**Comment by andreabassi78** [comment]
Yes, this is something already discussed in[ magicgui issue 239](https://github.com/napari/magicgui/issues/239), see comments at the bottom by @giorgia.
In practice the instance is used to share a variable between the 2 widgets. There should be a more practical way to do that, with a global variable. Working on that already.

**Comment by andreabassi78** [comment]
I removed the double instance which meant that we created a second widget in the backend. I changed this with [a dataclass](https://github.com/GiorgiaTortora/napari-roi-registration/blob/91423253f45dd59fbd9aed9847f7ce291796cb62/src/napari_roi_registration/_widget.py#L437), which looks more efficient. I ask @GiorgiaTortora to double-check that everything is working correctly so that we can close the issue.
Still not sure whether this is the most naparonic way to share variables between widgets, let's keep an eye on this.
